test/openssl/test_provider.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+require_relative 'utils'
+if defined?(OpenSSL) && defined?(OpenSSL::Provider)
+
+class OpenSSL::TestProvider < OpenSSL::TestCase
+  def test_openssl_provider_name_inspect
```

Comment:
I missed this last time, but this probably also needs to be skipped. IIRC, OpenSSL 3.0's FIPS module uses the `fips` provider to replace the `default` provider.

Perhaps we should skip the entire `OpenSSL::TestProvider`?